homeassistant/components/yeelight/light.py

```diff
@@ -287,124 +268,113 @@ def _lights_setup_helper(klass):
         _lights_setup_helper(YeelightGenericLight)
         _LOGGER.warning(
             "Cannot determine device type for %s, %s. Falling back to white only",
-            device.ipaddr,
+            device.ip_addr,
             device.name,
         )
 
-    hass.data[PLATFORM_DATA_KEY] += lights
     async_add_entities(lights, True)
-    await hass.async_add_executor_job(partial(setup_services, hass))
-
-
-def setup_services(hass):
-    """Set up the service listeners."""
-
-    def service_call(func):
-        def service_to_entities(service):
-            """Return the known entities that a service call mentions."""
-
-            entity_ids = extract_entity_ids(hass, service)
-            target_devices = [
-                light
-                for light in hass.data[PLATFORM_DATA_KEY]
-                if light.entity_id in entity_ids
-            ]
-
-            return target_devices
-
-        def service_to_params(service):
-            """Return service call params, without entity_id."""
-            return {
-                key: value
-                for key, value in service.data.items()
-                if key != ATTR_ENTITY_ID
-            }
-
-        def wrapper(service):
-            params = service_to_params(service)
-            target_devices = service_to_entities(service)
-            for device in target_devices:
-                func(device, params)
+    await _async_setup_services(hass)
 
-        return wrapper
 
-    @service_call
-    def service_set_mode(target_device, params):
-        target_device.set_mode(**params)
+async def _async_setup_services(hass: HomeAssistant):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside. Decorate it with `@callback`.